### ISSUE TITLE
First example in keras guide "customizing_what_happens_in_fit" does not work

### ISSUE THREAD

**Issue by pneumoman** [tensorflow/tensorflow#40740]
Thank you for submitting a TensorFlow documentation issue. Per our GitHub
policy, we only address code/doc bugs, performance issues, feature requests, and
build/installation issues on GitHub.

The TensorFlow docs are open source! To get involved, read the documentation
contributor guide: https://www.tensorflow.org/community/contribute/docs

## URL(s) with the issue:
https://www.tensorflow.org/guide/keras/customizing_what_happens_in_fit#a_first_simple_example

Please provide a link to the documentation entry, for example:
https://www.tensorflow.org/guide/keras/customizing_what_happens_in_fit#a_first_simple_example

## Description of issue (what needs changing):
The First Example doesn't seem to actually call **train_step()**

### Clear description
I have run this example using a debugger, and it does not actually get called
I have tested this with version 2.2.0

### Correct links
I have not been able to find anything that explains how this is supposed to work.

### Parameters defined
N/A

### Returns defined

N/A

### Raises listed and defined

Are the errors defined? For example,
N/A

Is there a usage example?

N/A

### Request visuals, if applicable

N/A
### Submit a pull request?

I can't tell if this should work on not.

**Comment by ravikyram** [comment]
@pneumoman 

I have tried executing tutorial in colab with TF 2.2 and i am not seeing any issue.Please, find the gist [here.](https://colab.research.google.com/gist/ravikyram/c6436f8bf6d8f20b3ae53b2713a67ff9/untitled62.ipynb).Can you please help me to understand where exactly you are facing the issue.If possible please share colab link or simple standalone code to reproduce the issue.It helps us in localizing the issue faster.Thanks!

**Comment by pneumoman** [comment]
You can't see what I'm talking about with just the standard collab page since it is just calling the normal loss function anyway. so I added some print statements to prove to myself that something was indeed being called.  And indeed it was however not as I expected, since it appears to have been called twice before the actual training and then no more, so I guess you need to be careful what kind of operations you put in there.
I found my problem, even though pip was reporting 2.2.0 for tensorflow, there was a reference to tensorflow-base==2.10, which after several install/uninstall go rounds, I was able to get it to reproduce the expected? output
